File: frontend/apps/readest-app/src/services/api/devices.ts
/**
 * 设备注册 API
 */

import { apiPost } from "./api";

export interface DeviceRegisterRequest {
  device_id: string;
  device_name?: string;
  platform?: string;
}

export interface DeviceInfo {
  device_id: string;
  device_name?: string;
  platform?: string;
  created_at: string;
}

export async function registerDevice(
  req: DeviceRegisterRequest,
): Promise<DeviceInfo> {
  const response = await apiPost<{ code: number; data: DeviceInfo }>(
    "/devices/register",
    req,
  );
  return response.data;
}

File: backend/app/api/v1/devices.py
"""设备注册接口"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.schemas.common import success, ApiResponse
from app.schemas.device import DeviceRegisterRequest, DeviceInfo
from app.services.device_service import DeviceService

router = APIRouter(prefix="/devices", tags=["设备管理"])


@router.post("/register", response_model=ApiResponse)
async def register_device(req: DeviceRegisterRequest, db: Session = Depends(get_db)):
    """注册或更新设备信息"""
    device = DeviceService.register(db, req)
    return success(data=DeviceInfo.model_validate(device).model_dump())
